ocs_ci/ocs/couchbase_new.py

```diff
@@ -0,0 +1,282 @@
+        """ "
+        Create secrets for running Couchbase workers
+
+        """
+        cb_secrets = templating.load_yaml(constants.COUCHBASE_NEW_WORKER_SECRET)
+        self.cb_secrets = OCS(**cb_secrets)
+        self.cb_secrets.create()
+        log.info("Successfully created secrets for Couchbase")
+        self.cb_create_cb_secret = True
+
+    def create_cb_cluster(self, replicas=1, sc_name=None):
+        """
+        Deploy a Couchbase server using Couchbase operator
+
+        Once the couchbase operator is running, we need to wait for the
+        worker pods to be up.  Once the Couchbase worker pods are up, pillowfight
+        task is started.
+
+        After the pillowfight task has finished, the log is collected and
+        analyzed.
+
+        Raises:
+            Exception: If pillowfight results indicate that a minimum performance
+                level is not reached (1 second response time, less than 1000 ops
+                per second)
+
+        """
+        logging.info("Creating Couchbase worker pods...")
+        cb_example = templating.load_yaml(constants.COUCHBASE_NEW_WORKER_EXAMPLE)
+        if storagecluster_independent_check():
+            cb_example["spec"]["volumeClaimTemplates"][0]["spec"][
+                "storageClassName"
+            ] = constants.DEFAULT_EXTERNAL_MODE_STORAGECLASS_RBD
+        cb_example["spec"]["servers"][0]["size"] = replicas
+        if sc_name:
+            cb_example["spec"]["volumeClaimTemplates"][0]["spec"][
+                "storageClassName"
+            ] = sc_name
+        self.cb_example = OCS(**cb_example)
+        self.cb_example.create()
+        self.cb_create_cb_cluster = True
+
+        # Wait for the Couchbase workers to be running.
+
+        logging.info("Waiting for the Couchbase pods to be Running")
+        self.pod_obj.wait_for_resource(
+            condition="Running",
+            selector="app=couchbase",
+            resource_count=replicas,
+            timeout=900,
+        )
+        log.info(
+            f"Expected number: {replicas} of couchbase workers reached running state"
+        )
+
+    def create_data_buckets(self):
+        """
+        Create data buckets
+
+        """
+        cb_bucket = templating.load_yaml(constants.COUCHBASE_NEW_DATA_BUCKET)
+        self.cb_bucket = OCS(**cb_bucket)
+        self.cb_bucket.create()
+        log.info("Successfully created data buckets")
+        self.cb_create_bucket = True
+
+    def run_workload(self, replicas, num_items=None, num_threads=None, run_in_bg=False):
+        """
+        Running workload with pillow fight operator
+        Args:
+            replicas (int): Number of pods
+            num_items (int): Number of items to be loaded to the cluster
+            num_threads (int): Number of threads
+            run_in_bg (bool) : Optional run IOs in background
+
+        """
+        self.result = None
+        logging.info("Running IOs using Pillow-fight")
+        if run_in_bg:
```

Comment:
I think we can get rid of this if:  doens't add much value considering it runs in bg by default, only the oc delay is what we see.

Reply:
run_pillowfights() reads the pillowfight yamls, waits for the pods to come up and wait till the pillowfight pods reaches completed state, this will take a good amount of time. This run_in_bg can be helpful for tests which needs a workload to run in background while the ocs operations are going on.  This is anyhow set to False by default and can be used whenever required in tests.

Reply:
:+1: